### PR TITLE
Multiple delegations to redoctober

### PR DESCRIPTION
**Pull request by ejcx**
The idea is to create a new type (to avoid ugly string parsing) and
then, instead of iterating through delegations with the username, iterate
through the delegations and look for your username and matching slot. Also
in cases we don't have the slot (everything but delegation), find the slot
when we match a user.

### CODE REVIEW

File: keycache/keycache.go [hunk 1]
```diff
@@ -78,76 +88,81 @@ func (usage Usage) matches(user string, labels []string) bool {
 }
 
 func NewCache() Cache {
```
Comment (kisom) [inline keycache/keycache.go]: Needs a comment.

File: keycache/keycache.go [hunk 2]
```diff
@@ -38,7 +48,7 @@ type ActiveUser struct {
 }
 
 type Cache struct {
```
Comment (kisom) [inline keycache/keycache.go]: `Cache` should also have a comment, even though it didn't previously --- now's a good time to add one.

File: keycache/keycache.go [hunk 3]
```diff
@@ -78,76 +88,81 @@ func (usage Usage) matches(user string, labels []string) bool {
...
 func NewCache() Cache {
-	return Cache{make(map[string]ActiveUser)}
+	return Cache{make(map[DelegateIndex]ActiveUser)}
 }
 
 // setUser takes an ActiveUser and adds it to the cache.
-func (cache *Cache) setUser(in ActiveUser, name string) {
-	cache.UserKeys[name] = in
+func (cache *Cache) setUser(in ActiveUser, name, slot string) {
+	d := &DelegateIndex{Name: name, Slot: slot}
```
Comment (kisom) [inline keycache/keycache.go]: Maybe `cache.UserKeys[DelegateIndex{Name: name, Slot: slot}) = in` ?

File: keycache/keycache.go [hunk 4]
```diff
@@ -78,76 +88,81 @@ func (usage Usage) matches(user string, labels []string) bool {
...
-	return
+	return false
 }
 
 // matchUser returns the matching active user if present
 // and a boolean to indicate its presence.
-func (cache *Cache) matchUser(name, user string, labels []string) (out ActiveUser, present bool) {
-	key, present := cache.UserKeys[name]
-	if present {
+func (cache *Cache) matchUser(name, user string, labels []string) (out ActiveUser, slot string, present bool) {
```
Comment (kisom) [inline keycache/keycache.go]: If the variables aren't going to be used, we should remove them from the function signature.

File: keycache/keycache.go (hunk 3 above)
Comment (haneric21) [inline keycache/keycache.go]: Hi, 
I'm just trying to understand the code and learn more about go; is there a reason why you want a pointer to the DelegateTuple and then pass in the dereferenced tuple in line 93?
Thanks

**Comment by kisom** [comment]
Please remove the checked in `redoctober` binary, too. It might be a good idea to add this to `.gitignore`.

File: keycache/keycache.go (hunk 3 above)
Comment (ejcx) [inline keycache/keycache.go]: No. Thanks for the catch. I'll remove the extra dereferencing. It was habit

**Comment by kisom** [comment]
Looks good, just needs the tests fixed now.

**Comment by kisom** [comment]
LGTM.

**Pull request merged**